src/unittests/line_intersection.py

```diff
@@ -0,0 +1,87 @@
+import unittest
+
+from src.simple_commander.utils.line_intersection import line_intersection
+
+
+class MainControllerTestCase(unittest.TestCase):
+    def setUp(self):
+        self.unit1 = {'width': 2}
+        self.unit2 = {'width': 2}
+
+    def tearDown(self):
+        del self
+
+    def test_intersection_1(self):
+        A = (1, 1)
+        B = (3, 2)
+        C = (1, 3)
+        D = (3, 1)
+        int_point = line_intersection((A, B), (C, D), round(self.unit1['width'] / 2), round(self.unit2['width'] / 2))
+        self.assertEqual(int_point, (2, 2), 'incorrect point for intersection')
+
+    def test_intersection_2(self):
+        A = (1, 7)
+        B = (5, 2)
+        C = (11, 7)
+        D = (6, 2)
+        int_point = line_intersection((A, B), (C, D), round(self.unit1['width'] / 2), round(self.unit2['width'] / 2))
+        self.assertEqual(int_point, (6, 2), 'incorrect point for intersection')
```

Comment:
'incorrect point for intersection' in all your cases. Seems this message should be only in case when func return None